botocore/signers.py

```diff
@@ -56,7 +57,11 @@ def __init__(self, service_name, region_name, signing_name,
         self._signing_name = signing_name
         self._signature_version = signature_version
         self._credentials = credentials
-        self._event_emitter = event_emitter
+
+        self._event_emitter = None
+        if event_emitter is not None:
```

Comment:
Off-line discussion: True. There shouldn't be a case when emitter=None. We will update 4 test cases who violate this rule.